### PR TITLE
CM-41217 - Implement SBT restore support for SCA

### PR DESCRIPTION
**Pull request by naftalicy**

### CODE REVIEW

File: cycode/cli/files_collector/sca/sbt/restore_sbt_dependencies.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from cycode.cli.files_collector.sca.base_restore_dependencies import BaseRestoreDependencies
+from cycode.cli.models import Document
+
+SBT_PROJECT_FILE_EXTENSIONS = ['sbt']
+SBT_LOCK_FILE_NAME = 'build.sbt.lock'
+
+
+class RestoreSbtDependencies(BaseRestoreDependencies):
+    def __init__(self, context: click.Context, is_git_diff: bool, command_timeout: int) -> None:
+        super().__init__(context, is_git_diff, command_timeout)
```
Comment (MarshalX) [inline cycode/cli/files_collector/sca/sbt/restore_sbt_dependencies.py:15]: looks like it is blank constructor which calls constructor from the base. no need to override it here

File: cycode/cli/files_collector/sca/sca_code_scanner.py [hunk 2]
```diff
@@ -132,10 +129,9 @@ def add_dependencies_tree_document(
 
 def restore_handlers(context: click.Context, is_git_diff: bool) -> List[BaseRestoreDependencies]:
     return [
-        RestoreGradleDependencies(context, is_git_diff, BUILD_GRADLE_DEP_TREE_TIMEOUT),
-        RestoreMavenDependencies(context, is_git_diff, BUILD_GRADLE_DEP_TREE_TIMEOUT),
-        RestoreNugetDependencies(context, is_git_diff, BUILD_NUGET_DEP_TREE_TIMEOUT),
```
Comment (MarshalX) [inline cycode/cli/files_collector/sca/sca_code_scanner.py:137]: why nuget and npm were deleted?

**Review by MarshalX — COMMENTED** [review]
good job!

File: cycode/cli/files_collector/sca/sca_code_scanner.py (hunk 2 above)
Comment (MarshalX) [reply cycode/cli/files_collector/sca/sca_code_scanner.py:137]: i got the answer internally

File: cycode/cli/files_collector/sca/sbt/restore_sbt_dependencies.py (hunk 1 above)
Comment (naftalicy) [reply cycode/cli/files_collector/sca/sbt/restore_sbt_dependencies.py]: fixed

**Review by MarshalX — APPROVED** [review]

File: cycode/cli/commands/scan/repository/repository_command.py [hunk 3]
```diff
@@ -49,7 +49,9 @@ def repository_command(context: click.Context, path: str, branch: str) -> None:
             progress_bar.update(ScanProgressBarSection.PREPARE_LOCAL_FILES)
 
             file_path = file.path if monitor else get_path_by_os(os.path.join(path, file.path))
-            documents_to_scan.append(Document(file_path, file.data_stream.read().decode('UTF-8', errors='replace')))
+            documents_to_scan.append(Document(file_path,
+                                              file.data_stream.read().decode('UTF-8', errors='replace'),
+                                              absolute_path= get_path_by_os(os.path.join(path, file.path))))
```
[the remark this answers is not in the document]
Comment (naftalicy) [reply cycode/cli/commands/scan/repository/repository_command.py]: #cycode_sast_ignore_here

**Review by MarshalX — APPROVED** [review]

**Pull request merged**